packages/ts/src/components/treemap/index.ts

```diff
@@ -0,0 +1,121 @@
+import { Selection } from 'd3-selection'
+import { hierarchy, treemap } from 'd3-hierarchy'
+import { group, max } from 'd3-array'
+import { scaleLinear } from 'd3-scale'
+import { ComponentCore } from 'core/component'
+import { SeriesDataModel } from 'data-models/series'
+import { getColor } from 'utils/color'
+import { getString, getNumber, isNumber } from 'utils/data'
+import { smartTransition } from 'utils/d3'
+import { TreemapConfigInterface, TreemapDefaultConfig } from './config'
+import { TreemapNode } from './types'
+
+import * as s from './style'
+
+export class Treemap<Datum> extends ComponentCore<Datum[], TreemapConfigInterface<Datum>> {
+  static selectors = s
+  protected _defaultConfig = TreemapDefaultConfig as TreemapConfigInterface<Datum>
+  public config: TreemapConfigInterface<Datum> = this._defaultConfig
+
+  datamodel: SeriesDataModel<Datum> = new SeriesDataModel()
+  tiles: Selection<SVGGElement, unknown, SVGGElement, unknown>
+
+  constructor (config?: TreemapConfigInterface<Datum>) {
+    super()
+    if (config) this.setConfig(config)
+    this.tiles = this.g.append('g')
+  }
+
+  _render (customDuration?: number): void {
+    const { config, datamodel: { data } } = this
+    const duration = isNumber(customDuration) ? customDuration : config.duration
+
+    if (!config.layers?.length) {
+      console.warn('Unovis | Treemap: No layers defined')
+      return
+    }
+
+    const layerAccessors = config.layers.map(layerAccessor => (i: number) => getString(data[i], layerAccessor, i))
+    const nestedData = group(data.keys(), ...layerAccessors as [(d: number) => string])
+
+    const rootNode = config.value !== undefined
+      ? hierarchy(nestedData).sum(index => typeof index === 'number' && getNumber(data[index], config.value, index))
```

Comment:
I'm curious, what will happen when `typeof index` is not a number?

Also you can use the `isNumber` function from utils here.

Reply:
I still need to circle back and test this code path. I'm thinking to add a new example that tests this path, compare that to what nested donut does, and truly get to the bottom of what this is supposed to do, and why that `typeof` check is even there. I suspect is has to do with using the output of `group` as the input to `hierarchy`, in which case the type of `data` is different for internal nodes (a 2-element array) and leaf nodes (the integer index - which I think this is checking for).